src/movie/movie.resolver.ts

```diff
@@ -1,16 +1,25 @@
-import { Resolver, Query, Args } from '@nestjs/graphql'
+import { Args, Parent, ResolveField, Resolver, Query } from '@nestjs/graphql'
 import { MovieService } from './movie.service'
+import { Movie } from 'src/graphql'
 
 @Resolver('Movie')
 export class MovieResolver {
   constructor(private readonly movieService: MovieService) {}
 
   @Query()
-  async movie(@Args('imdbId') imdbId: string) {
+  async movie(@Args('imdbId') imdbId: string): Promise<Movie> {
     if (imdbId) {
-      return this.movieService.getMovieByImdbId(imdbId)
+      return await this.movieService.getMovieByImdbId(imdbId)
     } else {
-      return this.movieService.getRandomMovie()
+      return (await this.movieService.getRandomMovies())[0]
```

Comment:
This will fail badly if we somehow don't get back what we expect.

Reply:
That's correct. What would you suggest? Adding a try/catch block too? Or do you prefer then/catch?

Reply:
Would be a nice addition. But I think we can also just leave it.

Reply:
Let's at least use then/catch :)